scene/gui/label.cpp

```diff
@@ -429,6 +432,8 @@ void Label::regenerate_word_cache() {
 				space_count = 0;
 			} else if ((i == xl_text.length() || current == '\n') && last != nullptr && space_count != 0) {
 				//in case there are trailing white spaces we add a placeholder word cache with just the spaces
+                prev_insert_newline = false;
```

Comment:
The indent of this line is incorrect.

Otherwise looks good. Please squash the commits and force push. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).